### PR TITLE
Adding a contributing guide; closes #13

### LINKED ISSUE #13: Need contributor guide
How does someone add a new service? How do we make this self-service for command additions, features? How much is upstream gophercloud?

### PR DESCRIPTION
**Pull request by kenperkins**
This is a first pass of a contributing guide, based on the work from the Flynn team at https://github.com/flynn/flynn/blob/master/CONTRIBUTING.md.

It makes some assumptions on code style, and certificate of origin, but it was go specific, so I thought it was a good start.

Discussion to follow.

- Fixes #13

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+Please familiarize yourself with the Contribution Guidelines and Project Roadmap before contributing.
+
+There are many ways to help `rack` besides contributing code:
+
+ - Fix bugs or file issues
+ - Improve the [documentation](https://github.com/jrperritt/rack/tree/master/docs)
+
+## Contributing Code
+
+Unless you are fixing a known bug, we **strongly** recommend discussing it with the core team via a GitHub issue or on [Freenode IRC](irc://chat.freenode.net/rackspace) before getting started to ensure your work is consistent with `rack`'s roadmap and architecture.
```
Comment (rgbkrk) [inline CONTRIBUTING.md]: Are we actually all on freenode watching over this? I'd rather people just file issues no matter what.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+All contributions are made via pull request. Note that **all patches from all contributors get reviewed**. After a pull request is made other contributors will offer feedback, and if the patch passes review a maintainer will accept it with a comment. When pull requests fail testing, authors are expected to update their pull requests to address the failures until the tests pass and the pull request merges successfully.
+
+At least one review from a maintainer is required for all patches (even patches from maintainers).
```
Comment (rgbkrk) [inline CONTRIBUTING.md]: :+1:

File: CONTRIBUTING.md (hunk 1 above)
Comment (kenperkins) [inline CONTRIBUTING.md]: FUUUUUUUUUU.

_I think you're right, actually_.

Comment (jnoller) [inline CONTRIBUTING.md]: :fish_cake:

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+
+Reviewers should leave a `LGTM` or `:+1:` comment once they are satisfied with the patch. If the patch was submitted by a maintainer with write access, the pull request should be merged by the submitter after review.
+
+## Code Style
+
+Please follow these guidelines when formatting source code:
+
+* Go code should match the output of `gofmt -s`
+* Shell scripts should adhere to the [Google Shell Style guide](https://google-styleguide.googlecode.com/svn/trunk/shell.xml)
```
Comment (rgbkrk) [inline CONTRIBUTING.md]: Our current `prep-travis-release.sh` needs to be adapted to this, particularly error messages on stderr (return codes are right though).

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+
+## Developer’s Certificate of Origin
+
+All contributions must include acceptance of the DCO:
```
Comment (rgbkrk) [inline CONTRIBUTING.md]: What the...?

I think you can delete this section.

Comment (kenperkins) [inline CONTRIBUTING.md]: Do we even have rackspace nomenclature here? can we remove this section entirely? It came straight from flynn.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+For legal reasons, no anonymous or pseudonymous contributions are accepted ([contact us](mailto:sdk-support@rackspace.com) if this is an issue).
+
+## Pull request procedure
+
+To make a pull request, you will need a GitHub account; if you are unclear on this process, see GitHub's documentation on [forking](https://help.github.com/articles/fork-a-repo) and [pull requests](https://help.github.com/articles/using-pull-requests). Pull requests should be targeted at the `master` branch. Before creating a pull request, go through this checklist:
+
+1. Create a feature branch off of `master` so that changes do not get mixed up.
+1. [Rebase](http://git-scm.com/book/en/Git-Branching-Rebasing) your local changes against the `master` branch.
+1. Run the full project test suite with the `go test ./...` (or equivalent) command and confirm that it passes.
+1. Run `gofmt -s` (if the project is written in Go).
```
Comment (rgbkrk) [inline CONTRIBUTING.md]: This should be `gofmt ./...`.

Also, the project is written in go so the parenthetical statement can be removed.

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+1. Accept the Developer's Certificate of Origin on all commits (see above).
+1. Ensure that each commit has a subsystem prefix (ex: `files: `).
+
+Pull requests will be treated as "review requests," and maintainers will give feedback on the style and substance of the patch.
+
+Normally, all pull requests must include tests that test your change. Occasionally, a change will be very difficult to test for. In those cases, please include a note in your commit message explaining why.
+
+## Communication
+
+We use the #rackspace IRC channel on [Freenode](irc://chat.freenode.net/rackspace). You are welcome to drop in and ask questions, discuss bugs, etc. You can [connect using webchat](https://webchat.freenode.net/?channels=rackspace) if you do not have an IRC client.
```
Comment (rgbkrk) [inline CONTRIBUTING.md]: Toss this or link to GitHub.

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+
+## Conduct
+
+Whether you are a regular contributor or a newcomer, we care about making this community a safe place for you and we've got your back.
+
+* We are committed to providing a friendly, safe and welcoming environment for all, regardless of gender, sexual orientation, disability, ethnicity, religion, or similar personal characteristic.
+* Please avoid using nicknames that might detract from a friendly, safe and welcoming environment for all.
+* Be kind and courteous. There is no need to be mean or rude.
+* We will exclude you from interaction if you insult, demean or harass anyone. In particular, we do not tolerate behavior that excludes people in socially marginalized groups.
+* Private harassment is also unacceptable. No matter who you are, if you feel you have been or are being harassed or made uncomfortable by a community member, please contact one of the channel ops or a member of the `rack` core team immediately.
```
Comment (rgbkrk) [inline CONTRIBUTING.md]: Remove mentions of chanops.

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+* Likewise any spamming, trolling, flaming, baiting or other attention-stealing behaviour is not welcome.
+
+We welcome discussion about creating a welcoming, safe, and productive environment for the community. If you have any questions, feedback, or concerns [please let us know](mailto:sdk-support@rackspace.com).
+
+## Attribution
+
+This contributing guide was directly based off of the [Flynn contributing guide](https://github.com/flynn/flynn/blob/master/CONTRIBUTING.md). Thanks to the Flynn team for providing a great example!
```
Comment (rgbkrk) [inline CONTRIBUTING.md]: :+1:

File: CONTRIBUTING.md (hunk 3 above)
Comment (kenperkins) [inline CONTRIBUTING.md]: Can you assist with that on this branch?

Comment (rgbkrk) [inline CONTRIBUTING.md]: Sure!

Comment (rgbkrk) [inline CONTRIBUTING.md]: Done.

**Comment by rgbkrk** [comment]
Ended up updating the environment variables according to Google's spec as well.

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -0,0 +1,58 @@
+# Contribution Guide
+
+We welcome and encourage community contributions to `rack`.
+
+Please familiarize yourself with the Contribution Guidelines and Project Roadmap before contributing.
+
+There are many ways to help `rack` besides contributing code:
+
+ - Fix bugs or file issues
```
Comment (rgbkrk) [inline CONTRIBUTING.md]: Is this meant to be "find bugs", since it's "...besides contributing code"

Comment (kenperkins) [inline CONTRIBUTING.md]: It was copied verbatim, but yes, that makes more sense.

**Comment by rgbkrk** [comment]
Still WIP? I think this is good enough to change the title and merge it.

**Comment by jrperritt** [comment]
:+1:

**Comment by kenperkins** [comment]
Good to go.

**Pull request merged**